### ISSUE TITLE
Access control policies don't work

### ISSUE THREAD

**Issue by sisumara** [netbirdio/netbird#1718]
**Describe the problem**

Today I've deployed a new setup which has separated IDP based on Zitadel and Netbird which I've configured using advanced configuration guide. I have one server in a cloud which has docker installation with couple of services and netbird agent installed in the same network with docker services. I've deleted default access policy rule and added custom one, which allows the only UDP traffic.
<img width="1496" alt="Screenshot 2024-03-15 at 4 54 41 PM" src="https://github.com/netbirdio/netbird/assets/61536902/1ae572f8-dc72-4b31-af60-d7770dd0f66c">
I've added a new route with peer and the docker network.
<img width="1496" alt="Screenshot 2024-03-15 at 4 56 58 PM" src="https://github.com/netbirdio/netbird/assets/61536902/0efa39b5-7852-466b-9b54-f91d4c5d05fe">

And traffic began to flow from User peer to OracleDC peer, but not the traffic which is limited by the access policy rule, but any traffic. So, in this case I allowed the only UDP traffic, but I can ping hosts in remote network and access to the services on 80,443/tcp ports. 

**To Reproduce**

1. Deploy a new service
2. Add some peers to different groups.
3. Create network route which sits behind one of the peers.
4. Delete default access policy and create a new one which limits the access to the peers with network routes.
5. Feel free to access to any resources in the route without limitations of created access policy.

**Expected behavior**

Limited traffic according to created Access Policy.

**Are you using NetBird Cloud?**

self-host NetBird's control plane.

**NetBird version**

`0.26.3`

Thank you

**Comment by braginini** [comment]
Hey @sisumara 
This is a know limitation. The access control rule you have set up is applied to the routing peer but not the machines behind it. 
We are working on enhancing the access control to apply port and protocol restrictions on the machines behind the router.

Do you want to limit ports and protocols for the whole network behind the router or individual resources?

**Comment by sisumara** [comment]
> Hey @sisumara This is a know limitation. The access control rule you have set up is applied to the routing peer but not the machines behind it. We are working on enhancing the access control to apply port and protocol restrictions on the machines behind the router.
> 

So, I can limit traffic flow with access policies the only to the peers, right? 

> Do you want to limit ports and protocols for the whole network behind the router or individual resources?

I have couple of cases and in some of them it needs to limit the whole network in some of them just certain hosts in the network.

**Comment by barto95100** [comment]
YEs the same, test to accept juste in TCP port 32400 (plex) and the host with netbird agent havec access to connecte on SSH :
explain flow 

host peer agent ----> host routing peer -----> server plex 

I want to block the host peer agent just right access on port 32400 and not all port in host beinhind routing peer

**Comment by Cheekie25** [comment]
I have the same need : I would like to limit ports and protocols for the whole network behind the router.
Do you any ETA for this feature ?
Do you know any workaround ?
Thank you

**Comment by nazarewk** [comment]
> I have the same need: I would like to limit ports and protocols for the whole network behind the router. Do you any ETA for this feature ? Do you know any workaround ? Thank you

I am pretty sure it's perfectly achievable and working with:
- introduction of protocols and ports on access policies
- the new Networks (made of Resources and Routing Peers) on the current version.

I am closing this because I believe this is fully addressed, but feel free to comment and we'll reopen.

**Comment by Cheekie25** [comment]
Thank you for your answer.

My goal is to be able to reach some web service on a specific port of a resource but that the other web services on the same resource do not be accessible.

I just tried the following setup without success:

- One docker peer "NAS" with network mode host (v0.41.3).
Groups "All" and "Secret".
- One iPhone (v0.36.4-dev according to the the UI and 0.0.9 in the app. No update is available in the App Store).
Group "All".
- One network named "Private" with one resource "MY NAS" with 192.168.1.50/32 for the address.
Group "Secret".
Routing peer: "NAS" (Masquerade activated).

I have two http web services running on "MY NAS".
One on port 9000 and one one port 82.

Now if I don't set any policy to the "Private" network, I can't reach any service of my "MY NAS" resource from the iPhone.
But if I add a policy that links the groups, "All" to "Secrets", I can reach all of the service.

The issue is that even if the policy has the only direction "All" to "Secrets" and that the port 9000 and TCP protocol are specified, I can still reach the same resource through the port 82.

I'm I missing something ?

**Comment by nazarewk** [comment]
> I'm I missing something ?

1. You should use a dedicated group (let's call it `X`) for your client devices instead of `All`.
2. Then create Access Control policy allowing access from Group `X` to Group `Secret`
3. limit the policy to ports 9000 & 82. Note you need to select either `TCP` or `UDP` to do that
4. (optionally) duplicate the rule if you need access to both TCP and UDP on those ports

**Comment by Cheekie25** [comment]
I tried several setups, here are 3 but I tried others, but in each case: either nothing can be reached, or everything can.
I haven't found any case where only the 9000 port of the resource would be reachable (and the 82 port wouldn't). 

1)
<table><thead>
  <tr>
    <th rowspan="2">Devices <br>and IPs</th>
  </tr>
  <tr>
    <th>iPhone <br>100.75.71.147<br>- </th>
    <th>NAS (Routing peer)<br>100.75.15.109<br>192.168.1.50</th>
    <th>MY NAS (Resource)<br>- <br>192.168.1.50/32</th>
  </tr></thead>
<tbody>
  <tr>
    <td>Groups</td>
    <td>X</td>
    <td>Secret</td>
    <td>Secret</td>
  </tr>
</tbody>
</table>

<table><thead>
  <tr>
    <th>Policies</th>
    <th>Result</th>
  </tr></thead>
<tbody>
  <tr>
    <td>X --&gt; Secret (TCP port 9000)</td>
    <td>iPhone can connect to 192.168.1.50:9000 and 192.168.1.50:82</td>
  </tr>
</tbody>
</table>

2.
<table><thead>
  <tr>
    <th rowspan="2">Devices <br>and IPs</th>
  </tr>
  <tr>
    <th>iPhone <br>100.75.71.147<br>- </th>
    <th>NAS (Routing peer)<br>100.75.15.109<br>192.168.1.50</th>
    <th>MY NAS (Resource)<br>- <br>192.168.1.50/32</th>
  </tr></thead>
<tbody>
  <tr>
    <td>Groups</td>
    <td>X</td>
    <td>Secret</td>
    <td>Res</td>
  </tr>
</tbody>
</table>

<table><thead>
  <tr>
    <th>Policies</th>
    <th>Result</th>
  </tr></thead>
<tbody>
  <tr>
    <td>X --&gt; Secret (TCP port 9000)</td>
    <th rowspan="2">iPhone can't connect to 192.168.1.50</th>
  </tr>
  <tr>
    <td>Secret --&gt; Res (TCP port 9000)</td>
  </tr>
</tbody>
</table>

3.
<table><thead>
  <tr>
    <th rowspan="2">Devices <br>and IPs</th>
  </tr>
  <tr>
    <th>iPhone <br>100.75.71.147<br>- </th>
    <th>NAS (Routing peer)<br>100.75.15.109<br>192.168.1.50</th>
    <th>MY NAS (Resource)<br>- <br>192.168.1.50/32</th>
  </tr></thead>
<tbody>
  <tr>
    <td>Groups</td>
    <td>X</td>
    <td>Secret</td>
    <td>Res</td>
  </tr>
</tbody>
</table>

<table><thead>
  <tr>
    <th>Policies</th>
    <th>Result</th>
  </tr></thead>
<tbody>
  <tr>
    <td>X --&gt; Res (TCP port 9000)</td>
    <td>iPhone can connect to 192.168.1.50:9000 and 192.168.1.50:82</td>
  </tr>
</tbody>
</table>

**Comment by Cheekie25** [comment]
I tried with a Routing peer on another device than the Resource one.
This way the policies behave as expected.
Any way to make that work when the Routing Peer and the Resource are on the same device ?

**Comment by jkolczasty** [comment]
Same here. I added e.g. policy to allow Group1 to some resource ICMP only and user can still connect to TCP ports. If I disable policy with ICMP, user no long can connect TCP connections. Looks like policy is working on L3 only.

Client: 0.65.1
Management: v0.65.1

Peer and Resource are NOT in the same network, not even in the same location.

See my comment: https://github.com/netbirdio/netbird/issues/4787#issuecomment-3908066932
